rtl/eclrtl/eclrtl.cpp

```diff
@@ -1287,16 +1310,22 @@ bool rtlDataToBool(unsigned len, const void * _src)
 
 void rtlBoolToData(unsigned tlen, void * tgt, bool src)
 {
-    memset(tgt, 0, tlen);
-    if (src)
-        ((char *)tgt)[tlen-1] = 1;
+    if (likely(tlen))
+    {
+        memset(tgt, 0, tlen);
+        if (src)
+            ((char *)tgt)[tlen-1] = 1;
```

Comment:
This would potentially have corrupted before, I think?

Reply:
I guess it would - if you had cast a boolean to a string0.

Reply:
Somebody must have done so in your suite for you to have caught this (assuming it was changed because you saw an issue from sanitize)

Reply:
When I hit an error with memset/memcmp/memcpy in any file I then searched the file for other potential issues.  This is almost certainly one of those.